src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -2112,6 +2603,35 @@ object Desugar {
       case PMethodReceiveName(typ)    => nameWithoutScope(s"$METHOD_PREFIX${typ.name}")(n, context)
       case PMethodReceivePointer(typ) => nameWithoutScope(s"P$METHOD_PREFIX${typ.name}")(n, context)
     }
+    private def stringifyType(typ: in.Type): String = typ match {
+      case _: in.BoolT => "Bool"
+      case in.IntT(_, kind) => s"Int${kind.name}"
+      case in.VoidT => ""
+      case _: in.PermissionT => "Permission"
+      case in.SortT => "Sort"
+      case in.ArrayT(len, elemT, _) => s"Array$len${stringifyType(elemT)}"
+      case in.SliceT(elemT, _) => s"Slice${stringifyType(elemT)}"
+      case in.SequenceT(elemT, _) => s"Sequence${stringifyType(elemT)}"
+      case in.SetT(elemT, _) => s"Set${stringifyType(elemT)}"
+      case in.MultisetT(elemT, _) => s"Multiset${stringifyType(elemT)}"
+      case in.OptionT(elemT, _) => s"Option${stringifyType(elemT)}"
+      case in.DefinedT(name, _) => s"Defined$name"
+      case in.PointerT(t, _) => s"Pointer${stringifyType(t)}"
+      case in.TupleT(ts, _) => s"Tuple${ts.map(stringifyType).mkString("")}"
```

Comment:
I do not know whether that this is particularly important, but this transformation is not injective.
`Tuple(Tuple(X), Y)` and `Tuple(Tuple(X, Y))` maps to the same.